js/adapt-contrib-accordion.js

```diff
@@ -34,33 +34,55 @@ define(function(require) {
 
         toggleItem: function(event) {
             event.preventDefault();
-            this.$('.accordion-item-body').stop(true, true).slideUp(200);
 
-            if (!$(event.currentTarget).hasClass('selected')) {
-                this.$('.accordion-item-title').removeClass('selected');
-                var body = $(event.currentTarget).addClass('selected visited').siblings('.accordion-item-body').slideToggle(200, function() {
-                  $(body).a11y_focus();
-                });
-                this.$('.accordion-item-title-icon').removeClass('icon-minus').addClass('icon-plus');
-                $('.accordion-item-title-icon', event.currentTarget).removeClass('icon-plus').addClass('icon-minus');
+            var toggleButton = $(event.currentTarget);
+            var accordionItem = toggleButton.parent('.accordion-item');
+            var isCurrentlyExpanded = toggleButton.hasClass('selected');
 
-                if ($(event.currentTarget).hasClass('accordion-item')) {
-                    this.setVisited($(event.currentTarget).index());
-                } else {
-                    this.setVisited($(event.currentTarget).parent('.accordion-item').index());
+            if (!this.model.get('_persistExpansion')) {
+                // Close and reset all Accordion items
+                var allAccordionItems = this.$('.accordion-item');
+                for (var i = 0; i < allAccordionItems.length; i++) {
```

Comment:
I doubt it's going to make much difference to performance here - but if you put the .length property as the loop condition it has to be checked on every iteration of the loop, so it's better to store it first - either like this:
```javascript
var count = allAccordionItems.length
for (var i = 0; i < count; i++) {
```
or like this if you prefer to keep it all on one line:
```javascript
for (var i = 0; count = allAccordionItems.length; i < count; i++) {
```
(don't feel obliged to change it here, this is more just FYI)